### ISSUE TITLE
Add Cross-Currency Payments

### ISSUE THREAD

**Issue by wojakkk** [XRPLF/xrpl-py#261]
Add functions that supports "cross-currency payments", this essentially means exchanging fiat-to-fiat using XRP,

Cross-currency payments that exchange two issued currencies automatically use XRP, when it decreases the cost of the payment, by connecting order books to deepen the pool of available liquidity. For example, a payment sending from USD to MXN automatically converts USD to XRP and then XRP to MXN if doing so is cheaper than converting USD to MXN directly.

If we really think about it, if **Alice** wants to exchange her MXN to JPY, there is a very small and a non-competitive orderbook since there is little to no demand for JPY/MXN, Alice can utilize cross-currency payments, which has more liquidity:

MXN -> XRP:  XRP -> JPY

Since there is liquidity in both MXN/XRP and JPY/XRP, Alice can exchange her MXN to JPY in seconds cheaply,

TLDR: Add cross-currency payment functions
Add function in: 
```
- xrpl.transaction.**cross_currency**
   > from_currency
       = The initial currency being used to exchange to "to_currency" using (from_currency)/XRP. 
   > to_currency
       = The currency chosen to receive from the cross-currency payment
```

Refer to: https://xrpl.org/cross-currency-payments.html

**Comment by ledhed2222** [comment]
hey there @wojakkk, adding helper functions for cross currency payments is in our backlog. we'll be likely tackling it after the 1.1 release (coming soon!). in the meantime, you can still use the full capabilities of the ledger to make cross currency payments by submitting `Payment` and `Offer` transactions directly, so you don't have to wait for any helper functions to do this:

```python
from xrpl.models.transactions import Payment
from xrpl.models.amounts import IssuedCurrencyAmount
from xrpl.transaction import safe_sign_and_submit_transaction
from xrpl.clients import JsonRpcClient
from xrpl.wallet import Wallet

wallet = Wallet(seed, 0)
client = JsonRpcClient(url)
safe_sign_and_submit_transaction(
  Payment(
    account=wallet.classic_address,
    amount=IssuedCurrencyAmount(
      currency="JPY",
      issuer="11231",  # you would need to know a JPY issuer to use
      value="100",
    ),
    send_max=IssuedCurrencyAmount(
      currency="MXN",
      issuer="11231",  # whatever the issuer is of the MXN you want to send
      value="1",
    ),
    destination="1231231",  # address of receiver
  ),
  wallet,
  client,
)
```

**Comment by wojakkk** [comment]
Thanks @ledhed2222, great to hear that its being developed soon!

**Comment by wojakkk** [comment]
@ledhed2222 So it this how the code that you linked me works? :

> amount var = the currency/amount being used in the transaction
> send_max var = the currency/amount that will be exchanged for amount_var and sent to destination address

Me -> "amount" currency  -> (exchanged for send_max, market price)-> "send_max" currency -> destination

How does this particularly work, im quite confused.

**Comment by ledhed2222** [comment]
You have it opposite. Check out the documentation for the payment transaction - https://xrpl.org/payment.html#main_content_wrapper especially the payment types section on cross currency payments - https://xrpl.org/payment.html#types-of-payments

**Comment by ledhed2222** [comment]
closing - please open another issue if you have more questions! feel free!
